### PR TITLE
<BadgeSelect/> - make component controlled, add missing tests and refactor

### PR DESCRIPTION
**Pull request by shlomitc**
1. set as controlled if `selectedId` prop is provided.
2. add a missing test for default behavior when `selectedId` isn't supplied

This change is backward compatible.
- if `selectedId` was not passed, it keeps behaving like the uncontrolled version
- if `selectedId` was passed, it must have changed the wrapping state for consistency

@aryelu - please verify that it answers your [issue](https://github.com/wix/wix-style-react/issues/2432)

### CODE REVIEW

File: src/BadgeSelect/BadgeSelect.spec.js [hunk 1]
```diff
@@ -86,21 +92,54 @@ describe('BadgeSelect', () => {
...
   });
 
-  it('should change badge skin after an option is selected', () => {
-    const {driver, badgeDriver} = createComponent();
-    const selectedIndex = 3;
 
-    driver.click();
-    driver.clickAtOption(selectedIndex);
-    expect(badgeDriver.getSkin()).toBe(options[selectedIndex].skin);
+  it('should should pick the first option if no selectedId given', () => {
```
Comment (lbelinsk) [inline src/BadgeSelect/BadgeSelect.spec.js]: double `should`

File: src/BadgeSelect/BadgeSelect.spec.js [hunk 2]
```diff
@@ -86,21 +92,54 @@ describe('BadgeSelect', () => {
...
+    const uncontrolledState = {selectedId: undefined};
+    const {badgeDriver} = createComponent(uncontrolledState);
```
Comment (lbelinsk) [inline src/BadgeSelect/BadgeSelect.spec.js]: createComponent receives `props`, so maybe rename the argument to uncontrolledProp?

File: src/BadgeSelect/BadgeSelect.spec.js (hunk 1 above)
Comment (shlomitc) [reply src/BadgeSelect/BadgeSelect.spec.js]: done

File: src/BadgeSelect/BadgeSelect.spec.js (hunk 2 above)
Comment (shlomitc) [reply src/BadgeSelect/BadgeSelect.spec.js]: done

**Pull request merged**